pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -0,0 +1,404 @@
+package usersignup
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/config"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"testing"
+)
+
+const (
+	nameMember = "east"
+	operatorNamespace = "toolchain-host-operator"
+)
+
+func TestReadUserApprovalPolicy(t *testing.T) {
+	r, _ := prepareReconcile(t, "test")
+
+	// Create a new ConfigMap
+	cmValues := make(map[string]string)
+	cmValues[config.ToolchainConfigMapUserApprovalPolicy] = config.UserApprovalPolicyAutomatic
+	cm := &v1.ConfigMap{
+		Data: cmValues,
+	}
+	cm.Name = config.ToolchainConfigMapName
+
+	_, err := r.clientset.CoreV1().ConfigMaps(operatorNamespace).Create(cm)
+	require.NoError(t, err)
+
+	policy, err := r.ReadUserApprovalPolicyConfig(operatorNamespace)
+	require.NoError(t, err)
+	require.Equal(t, config.UserApprovalPolicyAutomatic, policy)
+}
```

Comment:
Let's test all options:

- CM doesn't exist or it does but missing the expected key (the default one is used)
- The key set to manual approval
- The key set to auto-approval

**Update:** It looks like all the cases indirectly covered in the other tests. So, I guess it's fine not to duplicate them here.